Return error instead of panic when Invalid indicators present

JPrevost opened the pull request:
This was necessary due to our local record `002621436`.

Before the fix, we'd panic with an index out of range.

Now we pass this error and Mario can continue with the next record:
```
2019/01/14 11:37:07 Error parsing MARC record: 002621436
2019/01/14 11:37:07 Invalid Indicators detected
2019/01/14 11:37:07 {[{001 002621436} {005 20180629004505.0} {006 m     o  d        } {007 cr cnu|||unuuu} {008 180221s2018    nju     o     001 0 eng d} {    FMT [{K }]} {    020 [{a 9781119188421} {q (electronic bk. ;} {q oBook)}]} {    020 [{a 1119188423} {q (electronic bk. ;} {q oBook)}]} {    020 [{z 978119188391} {q (cloth)}]} {    020 [{z 9781119188391}]} {    020 [{z 1119188393}]} {7   024 [{a 10.1002/9781119188421} {2 doi}]} {    035 [{a (OCoLC)1023810884}]} {    035 [{a (OCoLC)1035062525}]} {    035 [{a (OCoLC)1035303176}]} {    040 [{a DG1} {b eng} {e rda} {e pn} {c DG1} {d OCLCF} {d NLE} {d OCLCQ} {d UAB} {d OCLCQ} {d UPM} {d YDX} {d OCLCO} {d STF}]} {  4 050 [{a RC857}]} {  4 060 [{a WI 800}]} {0 4 082 [{a 616.3/7} {2 23}]} {0 4 245 [{a The pancreas :} {b an integrated textbook of basic science, medicine, and surgery /} {c edited by Hans G. Beger [and more].}]} {    250 [{a Third edition.}]} {  1 264 [{a Hoboken, NJ :} {b Wiley Blackwell,} {c 2018.}]} {    300 [{a 1 online resource}]} {    336 [{a text} {b txt} {2 rdacontent}]} {    337 [{a computer} {b c} {2 rdamedia}]} {    338 [{a online resource} {b cr} {2 rdacarrier}]}] {99 97 109 32 73 105 32}}
```

gravesm commented on marc.go:
```diff
@@ -244,8 +244,12 @@ func NewMarcIterator(r io.Reader) *MarcIterator {
 	d.Tag = tag
-	d.Indicator1 = string(data[0])
-	d.Indicator2 = string(data[1])
+	if len(data) > 0 {
```
Should we actually check to make sure that it has at least 3 bytes of data? Just to avoid having to do this again when we come across a field with only one indicator.

JPrevost replied:
Probably. I wasn't really thinking about anything beyond "just @#!$ing work".

JPrevost replied:
I've obtained a single file with the problem record in it so we can build better tests around what is happening here. I'll see if I can come up with something better _with tests_ before we merge.

JPrevost commented:
@gravesm I think the latest commit addresses your comment about needing at least 3 bytes for a valid field. Please let me know if you were thinking of a different solution. Thanks!
